apps/webapp/src/state/staking.ts

```diff
@@ -158,21 +179,55 @@ export const createStakingSlice = (): SliceCreator<StakingSlice> => (set, get) =
       state.staking.votingPowerByValidatorInfo = getVotingPowerByValidatorInfo(validatorInfos);
     });
   },
-  loadUnstakedTokensByAccount: async () => {
+  loadUnstakedAndUnbondingTokensByAccount: async () => {
     const balancesByAccount = await getBalancesByAccount();
-    const unstakedTokensByAccount = balancesByAccount.reduce<Map<number, ValueView | undefined>>(
+    const { unstakedTokensByAccount, unbondingTokensByAccount } = balancesByAccount.reduce<{
+      unstakedTokensByAccount: Map<number, ValueView | undefined>;
+      unbondingTokensByAccount: Map<number, UnbondingTokensForAccount>;
+    }>(
       (prev, curr) => {
-        const unstakedTokens = curr.balances.find(
-          ({ balanceView }) => getDisplayDenomFromView(balanceView) === STAKING_TOKEN,
+        prev.unstakedTokensByAccount.set(
+          curr.index.account,
+          curr.balances.find(
+            ({ balanceView }) => getDisplayDenomFromView(balanceView) === STAKING_TOKEN,
+          )?.balanceView,
         );
```

Comment:
I find this a bit more readable unnested:

```typescript
const foundStakingToken = curr.balances.find(
   ({ balanceView }) => getDisplayDenomFromView(balanceView) === STAKING_TOKEN)?.balanceView;
prev.unstakedTokensByAccount.set(curr.index.account, foundStakingToken);
```

Reply:
Good call, done